tensorflow_probability/python/experimental/bijectors/highway_flow.py

```diff
@@ -0,0 +1,395 @@
+        parameters=parameters,
+        dtype=dtype,
+        name=name)
+
+  @property
+  def bias(self):
+    return self._bias
+
+  @property
+  def width(self):
+    return self._width
+
+  @property
+  def residual_fraction(self):
+    return self._residual_fraction
+
+  @property
+  def upper_diagonal_weights_matrix(self):
+    return self._upper_diagonal_weights_matrix
+
+  @property
+  def lower_diagonal_weights_matrix(self):
+    return self._lower_diagonal_weights_matrix
+
+  @property
+  def activation_fn(self):
+    return self._activation_fn
+
+  @property
+  def gate_first_n(self):
+    return self._gate_first_n
+
+  @property
+  def num_ungated(self):
+    return self._num_ungated
+
+  def _derivative_of_softplus(self, x):
+    return tf.concat([(self.residual_fraction) * tf.ones(
+      self.gate_first_n, dtype=self.dtype),
+                      tf.zeros(self.num_ungated, dtype=self.dtype)],
+                     axis=0) + (
+             tf.concat([(1. - self.residual_fraction) * tf.ones(
+               self.gate_first_n, dtype=self.dtype),
+                        tf.ones(self.num_ungated, dtype=self.dtype)],
+                       axis=0)) * tf.math.sigmoid(x)
+
+  def _convex_update(self, weights_matrix):
+    return tf.concat(
+      [self.residual_fraction * tf.eye(num_rows=self.gate_first_n,
+                                       num_columns=self.width,
+                                       dtype=self.dtype),
+       tf.zeros([self.num_ungated, self.width], dtype=self.dtype)],
+      axis=0) + tf.concat([(
+                               1. - self.residual_fraction) * tf.ones(
+      self.gate_first_n, dtype=self.dtype),
+                           tf.ones(self.num_ungated, dtype=self.dtype)],
+                          axis=0) * weights_matrix
+
+  def _inverse_of_softplus(self, y, n=20):
+    """Inverse of the activation layer with softplus using Newton iteration."""
+    x = tf.ones_like(y, dtype=self.dtype)
+    for _ in range(n):
+      x = x - (tf.concat([(self.residual_fraction) * tf.ones(
+        self.gate_first_n, dtype=self.dtype),
+                          tf.zeros(self.num_ungated, dtype=self.dtype)],
+                         axis=0) * x + tf.concat(
+        [(1. - self.residual_fraction) * tf.ones(
+          self.gate_first_n, dtype=self.dtype),
+         tf.ones(self.num_ungated, dtype=self.dtype)],
+        axis=0) * tf.math.softplus(
+        x) - y) / (
+            self._derivative_of_softplus(x))
+    return x
+
+  def _augmented_forward(self, x):
+    """Computes forward and forward_log_det_jacobian transformations.
+
+    Args:
+      x: Input of the bijector.
+
+    Returns:
+      x after forward flow and a dict containing forward and inverse log
+      determinant of the jacobian.
+    """
+
+    # Log determinant term from the upper matrix. Note that the log determinant
+    # of the lower matrix is zero.
+
+    fldj = tf.zeros(x.shape[:-1], dtype=self.dtype) + tf.reduce_sum(
```

Comment:
`x.shape` should be `ps.shape(x)` (also in the analogous line of `_augmented_inverse`)

Reply:
For the comments on the default behaviour, I have added a simple line for the HighwayFlow `__init__`, but maybe you meant something more sophisticated?